cardano-api/src/Cardano/Api/Eras.hs

```diff
@@ -365,6 +383,62 @@ instance IsShelleyBasedEra BabbageEra where
 instance IsShelleyBasedEra ConwayEra where
    shelleyBasedEra = ShelleyBasedEraConway
 
+data AnyShelleyBasedEra where
+  AnyShelleyBasedEra
```

Comment:
What problem is the introduction of this GADT solving?

Reply:
See https://github.com/input-output-hk/cardano-node/pull/5072/files#r1163416363